swupd/delta.go

```diff
@@ -0,0 +1,190 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// CreateDeltas creates all delta files between the previous and current
+// version of the supplied manifest, returning a list of files that
+// delta creation failed for.
+func CreateDeltas(manifest, statedir string, from, to uint32) ([]*File, error) {
+	var c config
+
+	c, err := getConfig(statedir)
+	if err != nil {
+		return nil, err
+	}
+
+	var oldManifest *Manifest
+	var newManifest *Manifest
+
+	if oldManifest, err = ParseManifestFile(filepath.Join(c.outputDir, fmt.Sprintf("%d", from), manifest)); err != nil {
+		return nil, err
+	}
+	if newManifest, err = ParseManifestFile(filepath.Join(c.outputDir, fmt.Sprintf("%d", to), manifest)); err != nil {
+		return nil, err
```

Comment:
Just a heads up: during integration with packs, I'll try to make this function (or a helper function) take the Manifest structs directly, since we already have all of them parsed. I might have to reset internal state used in linkPeers... before reusing though. (Let me know if this is a bad idea) No action needed for this PR.